annotator-core/src/main/java/edu/ucr/cs/riple/core/cache/BaseCache.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.core.cache;
+
+import com.google.common.collect.ImmutableSet;
+import edu.ucr.cs.riple.core.Config;
+import edu.ucr.cs.riple.core.metadata.index.Error;
+import edu.ucr.cs.riple.core.metadata.index.Fix;
+import edu.ucr.cs.riple.core.metadata.method.MethodDeclarationTree;
+import edu.ucr.cs.riple.injector.location.Location;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Base class for impact caches.
+ *
+ * @param <T> type of impacts saved in this model.
+ * @param <S> type of the map used to store impacts.
+ */
+public abstract class BaseCache<T extends Impact, S extends Map<Location, T>>
+    implements ImpactCache<T> {
+
+  /** Container holding cache entries. */
+  protected final S store;
+  /** Annotator config. */
+  protected final Config config;
+  /** Method declaration tree to store target module structure. */
+  protected final MethodDeclarationTree tree;
+
+  public BaseCache(Config config, S store, MethodDeclarationTree tree) {
+    this.store = store;
+    this.config = config;
+    this.tree = tree;
+  }
+
+  @Override
+  public boolean isUnknown(Fix fix) {
+    return !this.store.containsKey(fix.toLocation());
+  }
+
+  @Nullable
+  @Override
+  public T fetchImpact(Fix fix) {
+    return store.getOrDefault(fix.toLocation(), null);
```

Comment:
Thought it might throw exception as key not found (confused it with python dicts) but reading the javadoc it does not. 9ec4550fb3ee0962c445efaa904c72e49f8a5db0